src/app/BufferedReadCallback.cpp

```diff
@@ -215,19 +215,19 @@ CHIP_ERROR BufferedReadCallback::DispatchBufferedData(const ConcreteAttributePat
     //
     ReturnErrorOnFailure(reader.Next());
 
-    mCallback.OnAttributeData(mBufferedPath, &reader, statusIB);
+    mCallback.OnAttributeData(mBufferedPath, mDataVersion, &reader, statusIB);
 
     //
     // Clear out our buffered contents to free up allocated buffers, and reset the buffered path.
     //
     mBufferedList.clear();
     mBufferedPath = ConcreteDataAttributePath();
-
+    mDataVersion.ClearValue();
     return CHIP_NO_ERROR;
 }
 
-void BufferedReadCallback::OnAttributeData(const ConcreteDataAttributePath & aPath, TLV::TLVReader * apData,
-                                           const StatusIB & aStatus)
+void BufferedReadCallback::OnAttributeData(const ConcreteDataAttributePath & aPath, Optional<DataVersion> & aVersion,
```

Comment:
Followup: we should decide what we do if we have a chunked result come in and aVersion does not match mDataVersion.

Reply:
https://github.com/project-chip/connectedhomeip/issues/14530, yep